xsensmt/XsensMT.cpp

```diff
@@ -108,13 +108,29 @@ bool XsensMT::open(yarp::os::Searchable &config)
         yWarning() << "xsensmt -  Parameter \"frame_name\" not set. Using the same value as \"sensor_name\" for this parameter.";
     }
 
+    yDebug() << "xsensmt - test - xsensmt_period, check xsensmt_period: " << config.check("xsensmt_period");
+    yDebug() << "xsensmt - test - xsensmt_period, xsensmt_period is string: " << config.find("xsensmt_period").isString();
+    yDebug() << "xsensmt - test - xsensmt_period, xsensmt_period is int: " << config.find("xsensmt_period").isInt32();
+    yDebug() << "xsensmt - test - xsensmt_period, xsensmt_period is float: " << config.find("xsensmt_period").isFloat64();
+
+    if (config.check("xsensmt_period") && ( config.find("xsensmt_period").isInt32() || config.find("xsensmt_period").isFloat64()))
+    {
+        m_outputPeriod = config.find("xsensmt_period").asFloat64();
```

Comment:
Are you sure that this works as intended if xsensmt_period is a integer?

Reply:
I ran some checks, if I set `xsensmt_period` to an integer value (e.g. `2`), `config.find("xsensmt_period").isInt32()` returns `true` and `config.find("xsensmt_period").isFloat()` returns `false`. The opposite happens when I set a floating number (e.g. `2.0`)

`config.find("xsensmt_period").isString()` returned `false` in both cases.

That's why I put an `OR`.

On the other hand, `m_outputPeriod = config.find("xsensmt_period").asFloat64();` works as expected. If I set `2` for example, it's stored as `2.00`, and the float value is returned correctly too (with 2 decimal points).

Reply:
Perfect!